src/toast/instrument.py

```diff
@@ -423,6 +425,7 @@ class Focalplane(object):
         file (str):  Load the focalplane from this file.
         comm (MPI.Comm):  If loading from a file, optional communicator to broadcast
             across.
+        thinfp (int):  Only sample the detectors in the file.
```

Comment:
I agree that this seems like a generally useful feature.  The resulting "thinned" focalplane will depend on how an experiment has indexed their detectors, but if they want some other way of sampling a subset of detectors, then an experiment would provide those tools and not use this feature.